components/supervisor/pkg/gitpod/gitpod-config.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2020 TypeFox GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package gitpod
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ghodss/yaml"
+	"github.com/gitpod-io/gitpod/common-go/log"
+)
+
+// ConfigInterface provides an acess to the gitpod config file.
+type ConfigInterface interface {
+	// Observe provides channels triggered whenever the config is changed or errored
+	Observe(ctx context.Context) (<-chan *GitpodConfig, <-chan error)
+}
+
+// ConfigService provides an acess to the gitpod config file.
+type ConfigService struct {
+	location      string
+	locationReady <-chan struct{}
+
+	config    *GitpodConfig
+	listeners map[configListener]struct{}
+	stop      context.CancelFunc
+	mutex     sync.Mutex
+	timer     *time.Timer
```

Comment:
nit: I'd prefer a name that explains what the timer is for, e.g. `configPollTimer`